### ISSUE TITLE
config.xml default hostname issue

### ISSUE THREAD

**Issue by drshawnkwang** [BOINC/boinc#1528]
[https://boinc.berkeley.edu/trac/wiki/MakeProject](https://boinc.berkeley.edu/trac/wiki/MakeProject)

I was going through the makeproject instructions for setting up a devel system for BOINC. To that end I created a project with the test_app. When I went to start the project, I received the following error:
```
boinc@puppet-eah-agent:~/projects/eahdev$ ./bin/start
Entering ENABLED mode
Starting daemons
No daemons for this host found - check host name in config.xml

```

The FQDN of the computer is puppet-eah-agent.cgca.uwm.edu. That is what was in config.xml after running the make_project script. However, I needed to change this to the short domain name (puppet-eah-agent) for the daemons to start running.

Digging through the source revealed: tools/make_project, lines 23 & 167
```
NODENAME = getfqdn()
...
defopt('project_host'  , NODENAME, isdir=False)
```

While: sched/start, lines 655-657:
```
local_hostname = socket.gethostname()
local_hostname = local_hostname.split('.')[0]
```
In sched/start, local_hostname is compared to task.host. The latter is retrieved from the config.xml file.

Obviously the issue is that one uses the FQDN and the other the short domain. 

Finally, OS=linux Debian 8
version= git clone of BOINC/boinc, master branch

**Comment by ChristianBeer** [comment]
The make_project script should insert the short domain into config.xml but use the FQDN for scheduler.txt and index.html

I just spend 20 minutes looking for a pace to change but it is too tricky or too late for this. My guess is we could strip the domain name here: https://github.com/BOINC/boinc/blob/master/tools/make_project#L169 since it is only used to build the url_base and this is then used to build the cgi_url.

**Comment by drshawnkwang** [comment]
tools/make_project:169
```
# strip project_host to get short hostname
options.project_host=options.project_host.split('.')[0]
```

Per your suggestion I added the above two lines in make_project

I tested the script with:
```
./make_project -v --no_db --test_app --project_root /BOINC/projects/tah3 tah3 'Test3@home'
```
It generated the following config.xml (cut and paste)
```
<boinc>
    <config>
        <upload_dir>/BOINC/projects/tah3/upload</upload_dir>
        <send_result_abort>1</send_result_abort>
        <long_name>Test3@home</long_name>
        <sched_debug_level>3</sched_debug_level>
        <cache_md5_info>1</cache_md5_info>
        <upload_url>http://puppet-eah-agent.cgca.uwm.edu/tah3_cgi/file_upload_handler</upload_url>
        <disable_account_creation>0</disable_account_creation>
        <uldl_dir_fanout>1024</uldl_dir_fanout>
        <download_url>http://puppet-eah-agent.cgca.uwm.edu/tah3/download</download_url>
        <db_user>boinc</db_user>
        <log_dir>/BOINC/projects/tah3/log_puppet-eah-agent</log_dir>
        <app_dir>/BOINC/projects/tah3/apps</app_dir>
        <download_dir>/BOINC/projects/tah3/download</download_dir>
        <fuh_debug_level>3</fuh_debug_level>
        <master_url>http://puppet-eah-agent.cgca.uwm.edu/tah3/</master_url>
        <host>puppet-eah-agent</host>
        <db_name>tah3</db_name>
        <shmem_key>0x1111ff4e</shmem_key>
        <show_results>1</show_results>
        <key_dir>/BOINC/projects/tah3/keys/</key_dir>
        <dont_generate_upload_certificates>1</dont_generate_upload_certificates>
        <ignore_upload_certificates>1</ignore_upload_certificates>
        <db_passwd>

        </db_passwd>
        <min_sendwork_interval>6</min_sendwork_interval>
        <db_host>

        </db_host>
        <daily_result_quota>500</daily_result_quota>
        <one_result_per_user_per_wu>0</one_result_per_user_per_wu>
        <max_wus_to_send>50</max_wus_to_send>
    </config>
```

Thus it looks like **host** is the short hostname while other configs for URLs (such as **master_url**) are FQDN. 

The only thing that seems to use the short host name is **log_dir**.  In [https://github.com/BOINC/boinc/blob/master/py/Boinc/setup_project.py#L505](https://github.com/BOINC/boinc/blob/master/py/Boinc/setup_project.py#L505)
you can see that log_dir is normally to set log_FQDN. Adding the code above changes it to log_shortname. Will that cause problems?

Can you think of other tests I should do? (before making a pull-request?)

**Comment by ChristianBeer** [comment]
Looks fine to me. The log_shortname is the correct one. I've never seen it with the FQDN.

**Comment by drshawnkwang** [comment]
I'll initiate a pull request then.

**Comment by ChristianBeer** [comment]
fixed in b9a1338
